src/app.rb

```diff
@@ -20,18 +19,24 @@ class SinatraApp < Sinatra::Base
   register Sinatra::Shopify
   set :scope, 'read_products, read_orders'
 
-  register Sinatra::Partial
-  set :partial_template_engine, :erb
-  enable :partial_underscores
-
-  helpers Sinatra::ContentFor
+  register Kaminari::Helpers::SinatraHelpers
 
   # Home page
   get '/' do
     shopify_session do
       @shop = ShopifyAPI::Shop.current
       @charity = Charity.find_by(shop: current_shop_name)
-      @products = Product.where(shop: current_shop_name)
+      @products = Product.where(shop: current_shop_name).page(params[:products_page])
+      @donations = Donation.where(shop: current_shop_name).page(params[:donations_page])
+
+      @tab = if params[:products_page]
+        'products'
+      elsif params[:donations_page]
+        'donations'
+      else
+        'products'
+      end
```

Comment:
since I can't pass the `tab` param because of the issue I described below I implemented this solution. It works fine unless you try and switch between paginating products and orders without a refresh. Its not too bad though it just defaults to the products tab for each request. Also most users won't have enough products to even paginate.